content_handler/session_connection.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/content_handler/session_connection.h"
+#include "apps/mozart/lib/scene/session_helpers.h"
+
+namespace flutter_runner {
+
+SessionConnection::SessionConnection(mozart2::SessionPtr session,
+                                     mx::eventpair import_token)
+    : session_(std::move(session)), root_node_(&session_) {
+  root_node_.Bind(std::move(import_token));
+  session_.set_connection_error_handler(
+      std::bind(&SessionConnection::OnSessionError, this));
+  present_callback_ =
+      std::bind(&SessionConnection::OnPresent, this, std::placeholders::_1);
+}
+
+SessionConnection::~SessionConnection() {
+  session_.set_connection_error_handler(nullptr);
```

Comment:
there is no need to clear this, the session_ will go out of scope when the destructor returns and the handler will die with it